lib/fluent/plugin/in_http.rb

```diff
@@ -308,31 +274,37 @@ def on_server_connect(conn)
     def parse_params_default(params)
       if msgpack = params['msgpack']
         @parser_msgpack.parse(msgpack) do |_time, record|
-          return nil, record
+          if record.is_a?(Array)
+            # TODO: Temporarily supporting this case for compatibility.
+            # We should not consider this case here.
+            # We should fix MessagePackParser so that it doesn't return Array.
```

Comment:
I thought it should be summarized as fixing the JSON parser.
Since there is no time before the next release, I also wanted to keep the scope of impact as small as possible.
About `in_http`, I had no choice but to fix it in this PR.

If it looks like I should fix it in this PR, I will try today or tomorrow.
Or I can try to fix it in another PR after this PR is merged.

Reply:
It should be done in this PR instead of adding such workaround, since it's considered as a same problem and sometimes such workaround lives long unexpectedly.

I’ll postpone merging this to v1.17.0.

Reply:
Thanks for your review. I Understood.
My understanding was that the existing workaround was minimized here, not adding a new workaround.
But certainly, if it is not fixed entirely, it may lead to unnecessary problems.
Especially for `MessagePackParser` as `@custom_parser`.